Playground/SwiftShortcutsTemplate.playgroundbook/Contents/Modules/ShortcutSupport.playgroundmodule/Sources/ShortcutPreviewViewController.swift

```diff
@@ -0,0 +1,132 @@
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+
+        let button = UIButton(type: .custom)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addTarget(self, action: #selector(share(_:)), for: .touchUpInside)
+        button.setBackgroundImage(backgroundImage, for: .normal)
+        button.layer.masksToBounds = true
+        button.layer.cornerRadius = 4
+        button.addInteraction(UIDragInteraction(delegate: self))
+
+        button.addSubview(titleLabel)
+        view.addSubview(button)
+        view.backgroundColor = .secondarySystemBackground
+
+        NSLayoutConstraint.activate([
+            button.centerXAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerXAnchor),
+            button.centerYAnchor.constraint(equalTo: view.safeAreaLayoutGuide.centerYAnchor),
+            titleLabel.topAnchor.constraint(equalTo: button.topAnchor, constant: 6),
+            titleLabel.leadingAnchor.constraint(equalTo: button.leadingAnchor, constant: 8),
+            titleLabel.trailingAnchor.constraint(equalTo: button.trailingAnchor, constant: -8),
+            titleLabel.bottomAnchor.constraint(equalTo: button.bottomAnchor, constant: -6)
+        ])
+    }
+
+    @objc func share(_ sender: UIButton) {
+        do {
+            let fileManager = FileManager.default
+            let applicationSupport = try fileManager.url(
+                for: .applicationSupportDirectory,
+                in: .userDomainMask,
+                appropriateFor: nil,
+                create: false
+            )
+
+            let outputURL = applicationSupport
+                .appendingPathComponent("io.github.a2.swift-shortcuts", isDirectory: true)
+                .appendingPathComponent("\(name).shortcut", isDirectory: false)
+
+            // Remove the output directory if it already existed (clean up any old shortcuts)
+            if fileManager.fileExists(atPath: outputURL.deletingLastPathComponent().path) {
+                try fileManager.removeItem(at: outputURL.deletingLastPathComponent())
+            }
+
+            // Create the directory again
+            try fileManager.createDirectory(
+                at: outputURL.deletingLastPathComponent(),
+                withIntermediateDirectories: true,
+                attributes: nil
+            )
+
+            // Build the shortcut and write it to disk
+            let data = try shortcut.build()
+            try data.write(to: outputURL)
+
+            // Create an activity view controller pointing to the shortcut
+            let viewController = UIActivityViewController(
+                activityItems: [outputURL],
+                applicationActivities: nil
+            )
+
+            // Additionally configure iPad popover presentation if it's ever required
+            let controller = viewController.popoverPresentationController
+            controller?.sourceRect = sender.frame
+            controller?.sourceView = view
```

Comment:
We should make sure that the `sourceRect` is in the coordinate space of the `sourceView`:

```suggestion
            controller?.sourceRect = sender.bounds
            controller?.sourceView = sender
```

Reply:
It would be either way I think 🤔 

```swift
window.convert(sender.frame, from: view) == window.convert(sender.bounds, from: sender) // true
```

Assuming that `view === sender.superview` ... But I can change this to use `bounds` still 🙂